lib/src/backend.rs

```diff
@@ -137,6 +137,18 @@ content_hash! {
     }
 }
 
+impl Signature {
+    pub fn username(&self) -> Option<&str> {
+        // While this is not perfect (i.e., not exactly RFC 5322 compatible), it
+        // is very simple and should work well for a vast majority of the cases.
+        let parts: Vec<&str> = self.email.split('@').collect();
+        if let [out, _] = parts.as_slice() {
+            return Some(out);
+        };
+        None
```

Comment:
Should we make this function be `self.email.splitn(1, '@').next().unwrap()`, so we return the whole string if there's no "@" in the email? It's technically not a username then (or is it? I haven't checked the spec), but maybe it's what users are more likely to want to see? I'm not sure. Oh, I see you did that in the template. That also seems fine.